### PR TITLE
 feature + fix: add exception procession for dashscope_llm, add retry times (default 3 times)

### PR DESCRIPTION
**Pull request by Tlntin**
to close issue [232](https://github.com/modelscope/modelscope-agent/issues/232)

### CODE REVIEW

File: modelscope_agent/llm/dashscope_llm.py [hunk 1]
```diff
@@ -29,38 +30,60 @@ def generate(self,
...
+                            response)
+                        return llm_result
+                    else:
+                        err_msg = 'Error Request id: %s, Code: %d, status: %s, message: %s' % (
+                            response.request_id, response.status_code,
+                            response.code, response.message)
+                        print(err_msg)
+                        error_message_list.append(err_msg)
+                        time.sleep(i * 2 + 1)
+            except Exception as e:
```
Comment (wenmengzhou) [inline modelscope_agent/llm/dashscope_llm.py:93]: it is better to catch certain error and retry,  while some error should not, such as if a user forget to config dashscope api key, there is no need to retry

Comment (Tlntin) [reply modelscope_agent/llm/dashscope_llm.py:93]: you are right, I think 4xx status code may need not retry.
status code=500(when input length too long and inappropriate content) is same.

**Comment by wenmengzhou** [comment]
@zzclynn We may need to provide a general retry mechanism for all llm interface in our new framework,  we can discuss the solution and draw on the method of this pull request

**Comment by zzclynn** [comment]
@Tlntin thanks for your submission, we have upgrade modelscope-agent, and add a wrapper for retry all llms. 
i will close this  pull request, if you find other issues, we are very glad to recieve your pr again. 
Thanks, again.

**Pull request closed without merging**